### ISSUE TITLE
Muliples With joins

### ISSUE THREAD

**Issue by cchavez** [dimagi/django-cte#29]
Hi everyone.

I have the following scenario:

```
first_with = With(Model1, name="first_with")
second_with = With(Model2, name="second_with")
```
Then:

`first_with.join(Model0, id=first_with.col.id).with_cte(first_with)...`

That generate the SQL as expected like:

```
WITH RECURSIVE first_with AS ( ... )
SELECT <Model0 attributes>, <first_with attributes>
FROM <Molde0> join with <first_with> ...

```
So far good.

But then I want to add the second_with to the sql, but I was not able to add the join with Model0, so for example I tried:

`first_with.join(Model0, id=first_with.col.id).with_cte(first_with).with_cte(second_with)...`

That add the second_with to the WITH RECURSIVE list, but didn't make the join with Model0, I didn't figure out how to
do that. So for the sql looks like:

```
WITH RECURSIVE first_with AS ( ... ), second_with AS ( ... ) 
SELECT <Model0 attributes>, <first_with attributes>, <second_with attributes>
FROM <Model0> join with <first_with> ...

```
But second_with is not in the join with Model0, so the query fails. 

I tried to following the samples in the test cases, but I think there is no samples for my case.
Is it possible do to what I'm trying?

**Comment by millerdev** [comment]
Hi @cchavez, you have not specified how `first_with` is related to `second_with`, so I cannot construct an example. They need to be joined somehow. [`test_named_ctes`](https://github.com/dimagi/django-cte/blob/4be459b03bcdf147450c997430619631f2a76218/tests/test_cte.py#L246) has examples of how to do this, but that test seems a bit more complicated than your situation, so I added simpler test (`test_named_simple_ctes`) with multiple joins

https://github.com/dimagi/django-cte/blob/4be459b03bcdf147450c997430619631f2a76218/tests/test_cte.py#L210-L216

**Comment by cchavez** [comment]
@millerdev `first_with` and `second_with` are not related to each other, it's not a recursive with, they make two different calculation but I want to get both result in one single sql by join them with the Model0.

**Comment by cchavez** [comment]
@millerdev thank you for you help, I think I misunderstood the join method and I followed what to mentioned and it's working, it's generating the SQL that I expected.

I did the following:

`first_with.join(second_with.join(Modelo0, id=second_with.col.id), id=first_with.col.id).with_cte(first_with).with_cte(second_with)...`

And that generate the SQL that I expected, making the Model0 join with both second_with and first_with.

Than you very much for you help.
